test/test_bagitutils.py

```diff
@@ -78,24 +81,40 @@ def test_process_demo_data(self):
             self.assertNotIn('0', key)
 
     def test_fc_mock(self):
-        """Tests a mock file with a minimal set of columns."""
+        """Tests a small mock file with a minimal set of columns,
+        but which covers all use cases."""
         mock_simple = 'test/fc_mock.tsv'
-        real107 = '/home/michael/dev/manifest-handover/manifests/manifest_107_genomes.tsv'
-        with open(real107, 'r') as tsv:
+        with open(mock_simple, 'r') as tsv:
             lines = tsv.readlines()
         data = "\n".join(lines)
         bag = BagHandler(data=data, bag_info={}, bag_name='manifest')
-        participants, max_files_in_sample, protocols = bag.participants_and_max_files_in_sample_and_protocols()
-        # self.assertEqual(len(participants), 2)
-        # self.assertEqual(len(protocols), 1)
-        # self.assertEqual(max_files_in_sample, 4)
+
+        participants, max_files_in_sample, protocols = \
+            bag.participants_and_max_files_in_sample_and_protocols()
+        self.assertEqual(len(participants), 3)
+        self.assertEqual(len(protocols), 2)
+        self.assertEqual(max_files_in_sample, 2)
+
         samples = bag.samples(max_files_in_sample, protocols)
-        # self.assertEqual(len(samples), 1)
+        self.assertEqual(len(samples), 4)
 
-        print(bag.data)
-        data_path = '/home/michael/dev/manifest-handover/test_107'
-        bag.write_csv_files(data_path)
+        # Testing numbers of rows and columns in output samples.tsv file.
+        tmpdir = tempfile.mkdtemp()
+        bag.write_csv_files(tmpdir)
+        n_rows = 0
+        n_cols = 0
+        with open(tmpdir + '/sample.tsv', 'r') as fp:
+             reader = csv.reader(fp, delimiter='\t')
+             first_row = True
+             for row in reader:
+                 if first_row:
+                     first_row = False
+                     n_cols = len(row)  # number of columns in samples.tsv
+                 n_rows += 1
 
+        self.assertEqual(n_rows, 5)
```

Comment:
It might be a better/more thorough test if you checked in what the result should be, and then you just compare the two files. But this is optional.

Reply:
I agree. The TSV file lands as a list of lists, and we could just compare those.

Reply:
now compares the entired output from the bag.write_csv_file method to the one defined in setup